cli/command/service/create.go

```diff
@@ -40,10 +40,12 @@ func newCreateCommand(dockerCli *command.DockerCli) *cobra.Command {
 	flags.Var(&opts.constraints, flagConstraint, "Placement constraints")
 	flags.Var(&opts.networks, flagNetwork, "Network attachments")
 	flags.VarP(&opts.endpoint.ports, flagPublish, "p", "Publish a port as a node port")
+	flags.MarkHidden(flagPublish)
```

Comment:
I'm okay with keeping it visible, it's similar to `--volume` vs `--mount` on `docker run`, but happy to hear if we feel strongly about hiding it